lib/utils/createDomain.js

```diff
@@ -17,7 +18,7 @@ function createDomain(options, server) {
     hostname = '0.0.0.0';
   }
 
-  const port = server ? server.address().port : 0;
+  const port = server ? server.address().port : defaultPort;
```

Comment:
The intention was for client to use the `location` port: https://github.com/webpack/webpack-dev-server/blob/723e1ec394d2f1bdeda2e80dc8748b6da1c0a3a3/client-src/default/utils/createSocketUrl.js#L48-L50

Reply:
hmm yes missed this, @alexander-akait  better alternate?

Reply:
oh, good catch, we need refactor logic here, it is more complex